bundles/org.openhab.binding.blink/src/main/java/org/openhab/binding/blink/internal/handler/AccountHandler.java

```diff
@@ -0,0 +1,402 @@
+ * for the blink things camera and network.
+ *
+ * It also provides the methods for updating camera and network states.
+ *
+ * @author Matthias Oesterheld - Initial contribution
+ */
+@NonNullByDefault
+public class AccountHandler extends BaseBridgeHandler {
+
+    public static final String GENERATED_CLIENT_ID = "generatedClientId";
+    private static final OffsetDateTime EPOCH_UTC = Instant.EPOCH.atOffset(ZoneOffset.UTC);
+    private static final int TOKEN_TTL = 12; // hours
+    private final Logger logger = LoggerFactory.getLogger(AccountHandler.class);
+    BundleContext bundleContext;
+
+    @NonNullByDefault({})
+    AccountConfiguration config;
+    AccountService blinkService;
+    private final HttpService httpService;
+    private final NetworkAddressService networkAddressService;
+    private String generatedClientId = "";
+    @Nullable
+    AccountVerificationServlet accountServlet;
+    @Nullable
+    BlinkAccount blinkAccount;
+    @Nullable
+    BlinkHomescreen cachedHomescreen;
+    OffsetDateTime eventSince = EPOCH_UTC;
+    final Map<Long, BlinkEvents.Media> eventStore = new ConcurrentHashMap<>();
+    @Nullable
+    ScheduledFuture<?> refreshStateJob;
+    private final Object refreshStateMonitor = new Object();
+
+    public AccountHandler(Bridge bridge, HttpService httpService, BundleContext bundleContext,
+            NetworkAddressService networkAddressService, HttpClientFactory httpClientFactory, Gson gson) {
+        super(bridge);
+        this.httpService = httpService;
+        this.bundleContext = bundleContext;
+        this.networkAddressService = networkAddressService;
+        this.blinkService = new AccountService(httpClientFactory.getCommonHttpClient(), gson);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Collections.singleton(BlinkDiscoveryService.class);
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(AccountConfiguration.class);
+
+        // set the thing status to UNKNOWN temporarily and let the background task decide for the real status.
+        // the framework is then able to reuse the resources from the thing handler initialization.
+        // we set this upfront to reliably check status updates in unit tests.
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_PENDING, "Wait for login");
+
+        scheduler.execute(() -> {
+
+            // register 2FA Verification Servlet for this thing
+            if (accountServlet == null) {
+                try {
+                    logger.debug("Registering 2FA servlet");
+                    accountServlet = new AccountVerificationServlet(httpService, bundleContext, this, blinkService);
+                } catch (IllegalStateException e) {
```

Comment:
Unchecked exceptions shouldn't be caught. You could define a custom Exception inheriting from `Exception`.